onnxruntime/core/external_ops/pyop.h

```diff
@@ -0,0 +1,239 @@
+        }
+    }
+
+    // Do nothing since Custom Op does not trigger share inference
+    void GetOutputShape(OrtKernelContext*, size_t, OrtTensorTypeAndShapeInfo*) {}
+
+    void Compute(OrtKernelContext* context) {
+
+        ORT_ENFORCE (nullptr != context);
+        auto inputs_count = (size_t)reinterpret_cast<onnxruntime::OpKernelContextInternal*>(context)->InputCount();
+        std::vector<const void*>            inputs,      outputs;
+        std::vector<int32_t>                inputs_type, outputs_elem_size;
+        std::vector<std::vector<int64_t>>   inputs_dim,  outputs_dim;
+
+        for (size_t i = 0; i < inputs_count; ++i) {
+            auto ort_value = ort_.KernelContext_GetInput(context, i);
+            inputs.push_back(((MLValue*)ort_value)->Get<Tensor>().DataRaw());
+            inputs_type.push_back(GetType(ort_value));
+            inputs_dim.push_back(((MLValue*)ort_value)->Get<Tensor>().Shape().GetDims());
+        }
+
+        std::string err;
+        ORT_ENFORCE (PythonWrapper::GetInstance().invoke(instance_, compute_.c_str(), inputs, inputs_type, inputs_dim, outputs, outputs_elem_size, outputs_dim, logging_func_), PythonWrapper::GetInstance().lastErr(err));
+        for (size_t i = 0; i < outputs.size(); ++i) {
+            auto ort_output  = ort_.KernelContext_GetOutput(context, i, outputs_dim[i].data(), outputs_dim[i].size());
+            auto output_mem_addr = ort_.GetTensorMutableData<char>(ort_output);
+            auto output_len = std::accumulate(begin(outputs_dim[i]), end(outputs_dim[i]), static_cast<int64_t>(outputs_elem_size[i]), std::multiplies<int64_t>());
+            memcpy(output_mem_addr, outputs[i], output_len);
+            free(const_cast<void*>(outputs[i]));
+        }
+    }
+
+    int32_t GetType(const OrtValue* input) const
+    {
+        int32_t numpy_type;
+        ORT_ENFORCE (nullptr != input);
+        ORT_ENFORCE(((MLValue*)input)->IsTensor(), "input must be a tensor");
+        auto data_type = ((MLValue*)input)->Get<Tensor>().DataType();
+        if (data_type == DataTypeImpl::GetType<bool>()) {
+            numpy_type = 0;
+        } else if (data_type == DataTypeImpl::GetType<int8_t>()) {
+            numpy_type = 1;
+        } else if (data_type == DataTypeImpl::GetType<uint8_t>()) {
+            numpy_type = 2;
+        } else if (data_type == DataTypeImpl::GetType<int16_t>()) {
+            numpy_type = 3;
+        } else if (data_type == DataTypeImpl::GetType<uint16_t>()) {
+            numpy_type = 4;
+        } else if (data_type == DataTypeImpl::GetType<int32_t>()) {
+            numpy_type = 5;
+        } else if (data_type == DataTypeImpl::GetType<uint32_t>()) {
+            numpy_type = 6;
+        } else if (data_type == DataTypeImpl::GetType<int64_t>()) {
+            numpy_type = 9;
+        } else if (data_type == DataTypeImpl::GetType<uint64_t>()) {
+            numpy_type = 10;
+        } else if (data_type == DataTypeImpl::GetType<float>()) {
```

Comment:
I don't know how the compiler handles it, I know it is less readable, but for speed, most common types should be moved at the top. The gain would probably not significant though.

Reply:
Compiler has all sorts ways to optimize this function since it is simple if-else with constants.
With map compiler will be much less confident to generate efficient code.